metrics/web/docs/features/firebase_analytics/diagrams/firebase_analytics_domain_class.puml

```diff
@@ -0,0 +1,42 @@
+@startuml
+package base.domain.usecases {
+    interface UseCase<Type, Params> {
+        call: Type
+    }
+}
+
+package auth.domain {
+    package repositories {
+        abstract class AnalyticsRepository {
+            Future<void> logPageView()
+            Future<void> logLogin()
+        }
+    }
+    package usecases {
+        package parameters {
+            class UserIdParam {
+                userId : String
+            }
+
+            class PageNameParam {
+                name : String
```

Comment:
Would it be great to create a class that will be something like a custom enum that will store all available page names?